File: places.py
class Place(object):

    def __init__(self, name):
        """
        args: name: string
        """
        self.name = name
        self.connections = set()

    def __str__(self):
        return self.name


class Places(object):
    """
    instance of Places is contained in the state object (under state.places)
    Places represents the state of the places in a given world 
    """

    def __init__(self):
        self.places_dict = { 
            "arctic" : Place("the Arctic"),
            "cave" : Place("a dark cave"),
            "church" : Place("the church"),
            "countryside" : Place("the countryside"),
            "dark_alley" : Place("a dark alley"),
            "docks" : Place("the docks"),
            "lord_bartholomews_manor" : Place("Lord Bartholomew's manor"),
            "lord_carlos_manor" : Place("Lord Carlos' manor"),
            "market" : Place("the market"),
            "mermaid_rock" : Place("the mermaid rocks"),
            "ocean" : Place("the ocean"),
            "pirate_ship" : Place("a pirate ship"),
            "prison" : Place("prison"),
            "streets" : Place("the streets"),
            "tavern" : Place("the tavern"),
            "tower" : Place("the tower"),
            "upstairs" : Place("the upstairs of the tavern"),
            "void" : Place("the void"),
            "wizards_lab" : Place("the wizard's lab"),
            "woods" : Place("the woods"),
            }
        self.locked = set([
            self.places_dict["cave"],
            self.places_dict["prison"],
            self.places_dict["pirate_ship"],
            self.places_dict["ocean"],
            self.places_dict["void"]])
        self.burnable = set([
            self.places_dict["church"],
            self.places_dict["docks"],
            self.places_dict["lord_bartholomews_manor"],
            self.places_dict["lord_carlos_manor"],
            self.places_dict["market"],
            self.places_dict["tavern"],
            self.places_dict["tower"],
            self.places_dict["wizards_lab"],
            self.places_dict["woods"],
            ])
        self.burned = set()
        self.trashed = set()
        self.populated = frozenset([
            self.places_dict["docks"],
            self.places_dict["lord_bartholomews_manor"],
            self.places_dict["lord_carlos_manor"],
            self.places_dict["market"],
            self.places_dict["streets"],
            self.places_dict["tavern"],
            ])
        self.outside = frozenset([
            self.places_dict["arctic"],
            self.places_dict["cave"],
            self.places_dict["countryside"],
            self.places_dict["dark_alley"],
            self.places_dict["docks"],
            self.places_dict["market"],
            self.places_dict["mermaid_rock"],
            self.places_dict["ocean"],
            self.places_dict["pirate_ship"],
            self.places_dict["streets"],
            self.places_dict["woods"],
            ])
        self.inside = frozenset([
            self.places_dict["church"],
            self.places_dict["lord_bartholomews_manor"],
            self.places_dict["lord_carlos_manor"],
            self.places_dict["prison"],
            self.places_dict["tavern"],
            self.places_dict["tower"],
            self.places_dict["upstairs"],
            self.places_dict["wizards_lab"],
            ])
        self.town = frozenset([
            self.places_dict["church"],
            self.places_dict["dark_alley"],
            self.places_dict["docks"],
            self.places_dict["market"],
            self.places_dict["prison"],
            self.places_dict["streets"],
            self.places_dict["tavern"],
            self.places_dict["tower"],
            self.places_dict["upstairs"],
            self.places_dict["wizards_lab"],
            ])

        self.places_dict["arctic"].connections.add(self.places_dict["ocean"])
        self.places_dict["church"].connections.add(self.places_dict["market"])
        self.places_dict["church"].connections.add(self.places_dict["streets"])
        self.places_dict["countryside"].connections.add(self.places_dict["docks"])
        self.places_dict["countryside"].connections.add(self.places_dict["lord_bartholomews_manor"])
        self.places_dict["countryside"].connections.add(self.places_dict["streets"])
        self.places_dict["countryside"].connections.add(self.places_dict["woods"])
        self.places_dict["dark_alley"].connections.add(self.places_dict["streets"])
        self.places_dict["docks"].connections.add(self.places_dict["streets"])
        self.places_dict["docks"].connections.add(self.places_dict["countryside"])
        self.places_dict["lord_bartholomews_manor"].connections.add(self.places_dict["countryside"])
        self.places_dict["lord_carlos_manor"].connections.add(self.places_dict["woods"])
        self.places_dict["market"].connections.add(self.places_dict["streets"])
        self.places_dict["market"].connections.add(self.places_dict["church"])
        self.places_dict["market"].connections.add(self.places_dict["wizards_lab"])
        self.places_dict["mermaid_rock"].connections.add(self.places_dict["ocean"])
        self.places_dict["ocean"].connections.add(self.places_dict["arctic"])
        self.places_dict["ocean"].connections.add(self.places_dict["docks"])
        self.places_dict["ocean"].connections.add(self.places_dict["mermaid_rock"])
        self.places_dict["ocean"].connections.add(self.places_dict["pirate_ship"])
        self.places_dict["streets"].connections.add(self.places_dict["church"])
        self.places_dict["streets"].connections.add(self.places_dict["countryside"])
        self.places_dict["streets"].connections.add(self.places_dict["dark_alley"])
        self.places_dict["streets"].connections.add(self.places_dict["docks"])
        self.places_dict["streets"].connections.add(self.places_dict["market"])
        self.places_dict["streets"].connections.add(self.places_dict["tavern"])
        self.places_dict["streets"].connections.add(self.places_dict["tower"])
        self.places_dict["tavern"].connections.add(self.places_dict["streets"])
        self.places_dict["tower"].connections.add(self.places_dict["streets"])
        self.places_dict["upstairs"].connections.add(self.places_dict["tavern"])
        self.places_dict["wizards_lab"].connections.add(self.places_dict["market"])
        self.places_dict["woods"].connections.add(self.places_dict["countryside"])
        self.places_dict["woods"].connections.add(self.places_dict["lord_carlos_manor"])

File: web_version/places.js
"use strict";

exports.places = {
    "arctic": 
    {
        "burnable":  false,
        "links":     ["ocean"],
        "locked":    false,
        "name":      "the Arctic",
        "outside":   true,
        "populated": false,
        "town":      false,
        "trashable": false,
    },
    "cave": 
    {
        "burnable":  false,
        "links":     ["ocean"],
        "locked":    true,
        "name":      "a dark cave",
        "outside":   true,
        "populated": false,
        "town":      false,
        "trashable": false,
    },
    "church": 
    {
        "burnable":  true,
        "links":     ["market", "streets"],
        "locked":    false,
        "name":      "the church",
        "outside":   false,
        "populated": true,
        "town":      true,
        "trashable": false,
    },
    "countryside": 
    {
        "burnable":  false,
        "links":     ["lord_bartholomew_manor", "streets", "woods"],
        "locked":    false,
        "name":      "the countryside",
        "outside":   true,
        "populated": true,
        "town":      false,
        "trashable": false,
    },
    "dark_alley": 
    {
        "burnable":  false,
        "links":     ["streets"],
        "locked":    false,
        "name":      "a dark alley",
        "outside":   true,
        "populated": true,
        "town":      true,
        "trashable": false,
    },
    "docks": 
    {
        "burnable":  false,
        "links":     ["market", "ocean", "streets", "woods"],
        "locked":    false,
        "name":      "the docks",
        "outside":   true,
        "populated": true,
        "town":      true,
        "trashable": false,
    },
    "lord_bartholomew_manor": 
    {
        "burnable":  true,
        "links":     ["countryside"],
        "locked":    false,
        "name":      "Lord Bartholomew's Manor",
        "outside":   false,
        "populated": true,
        "town":      false,
        "trashable": false,
    },
    "lord_carlos_manor": 
    {
        "burnable":  true,
        "links":     ["woods"],
        "locked":    false,
        "name":      "Lord Carlos' Manor",
        "outside":   false,
        "populated": true,
        "town":      false,
        "trashable": true,
    },
    "market": 
    {
        "burnable":  true,
        "links":     ["church", "streets", "docks", "wizard_lab"],
        "locked":    false,
        "name":      "the market",
        "outside":   true,
        "populated": true,
        "town":      true,
        "trashable": true,
    },
    "mermaid_rock":
    {
        "burnable":  false,
        "links":     ["ocean"],
        "locked":    false,
        "name":      "the docks",
        "outside":   true,
        "populated": false,
        "town":      false,
        "trashable": false,
    },
    "ocean": 
    {
        "burnable":  false,
        "links":     [],
        "locked":    true,
        "name":      "the ocean",
        "outside":   true,
        "populated": false,
        "town":      false,
        "trashable": false,
    },
    "pirate_ship": 
    {
        "burnable":  false,
        "links":     [],
        "locked":    true,
        "name":      "Lord Arthur's Pirate Ship",
        "outside":   true,
        "populated": true,
        "town":      false,
        "trashable": false,
    },
    "prison": 
    {
        "burnable":  false,
        "links":     [],
        "locked":    true,
        "name":      "the docks",
        "outside":   false,
        "populated": true,
        "town":      false,
        "trashable": false,
    },
    "streets":
    {
        "burnable":  false,
        "links":     ["church", "countryside", "market", "tavern", "tower"],
        "locked":    false,
        "name":      "the streets",
        "outside":   true,
        "populated": true,
        "town":      true,
        "trashable": false,
    },
    "tavern":
    {
        "burnable":  true,
        "links":     ["streets"],
        "locked":    false,
        "name":      "the tavern",
        "outside":   false,
        "populated": true,
        "town":      true,
        "trashable": false,
    },
    "tower":
    {
        "burnable":  true,
        "links":     ["streets"],
        "locked":    false,
        "name":      "the tower",
        "outside":   false,
        "populated": true,
        "town":      true,
        "trashable": false,
    },
    "upstairs":
    {
        "burnable":  false,
        "links":     ["tavern"],
        "locked":    false,
        "name":      "the docks",
        "outside":   false,
        "populated": false,
        "town":      false,
        "trashable": true,
    },
    "void": 
    {
        "burnable":  false,
        "links":     [],
        "locked":    true,
        "name":      "the void",
        "outside":   false,
        "populated": false,
        "town":      false,
        "trashable": false,
    },
    "wizard_lab": 
    {
        "burnable":  true,
        "links":     ["market"],
        "locked":    false,
        "name":      "the wizard's lab",
        "outside":   false,
        "populated": false,
        "town":      false,
        "trashable": true,
    },
    "woods": 
    {
        "burnable":  true,
        "links":     ["countryside", "lord_carlos_manor", "ocean", "docks"],
        "locked":    false,
        "name":      "the woods",
        "outside":   true,
        "populated": false,
        "town":      false, 
        "trashable": false,
    },
};
